daft/execution/execution_step.py

```diff
@@ -82,6 +82,9 @@ def __str__(self) -> str:
     def __repr__(self) -> str:
         return self.__str__()
 
+    def name(self) -> str:
+        return f"{'-'.join(i.__class__.__name__ for i in self.instructions)}:{self.stage_id}"
```

Comment:
Nit: As a user, I might not know what the e.g. `:3` indicates at the end of a stage name; what do you think about making it `:stage-3` or something like that to indicate that it's a stage number?

Reply:
yeah sounds good